### ISSUE TITLE
Makes traveler's backback oofs on death

### ISSUE THREAD

**Issue by AnomalousDeity** [B1n-ry/Youre-in-grave-danger#97]
**Describe the bug**
when using the mod traveler's backpack, while wearing and if you die the pack straight up dies, it isnt in the grave nor is it placed

**To Reproduce**
1. equip backpack on ur back
2. die
3. aaand your bag is a goner

**Expected behavior**
the backpack to be placed nearby or be in the grave

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Minecraft Version: 1.19.2
 - Version: 1.4.6

**Additional context**

**Screenshots**

**Comment by B1n-ry** [comment]
Using trinket integration? Try to update the mod. Should be fixed in latest release. If it still happens in 1.4.7, let me know

**Comment by AnomalousDeity** [comment]
It still happens, on 1.4.7 :(

**Comment by B1n-ry** [comment]
And you're using latest travelers backpack version too? (8.2.10). The travelers backpack dev apparently fixed some sort of incompat with this mod in that version. Otherwise I can't really reproduce the issue myself with both latest versions of the mods

**Comment by neonishere** [comment]
Not OP but same problem, I updated both mods to the latest version and enabled all configs related to the issue on both mods :(

**Comment by B1n-ry** [comment]
Could you provide which configs (and possibly other mods) you're using?

**Comment by B1n-ry** [comment]
I'm going to update the mod later today (I think). If anyone has replied with reproducable steps, or steps I can follow to get a copy of your instance, so that the issue should happen to me too, I can probably fix the issue before I update
If not you would have to wait longer for a fix

**Comment by neonishere** [comment]
Since I needed a fast fix I just enabled the save trinkets on death config instead. Before that the backpack would just disappear from there and never come back to the trinket slot on pickup. The only things I enabled are the compat configs on both sides. I also thought that it might be because the backpack tries to place itself, but that config is disabled.

I'll try to get on to reproduce it in the following days, sorry I couldn't provide more info

**Comment by B1n-ry** [comment]
Favourably I'd like slightly more concrete information like a copy of the configuration files, or a full mod list or logs or whatever. Thanks for the update however. I appreciate you taking the time to find a way to reproduce the issue. I'm looking forward to hearing from you again in a few days or so then!

**Comment by neonishere** [comment]
EDIT: You can skip to the last comment

I tried to recreate the issue with current mods:

architectury-6.3.56-fabric
cloth-api-4.0.65
cloth-config-8.2.88-fabric
fabric-api-0.72.0+1.19.2
RoughlyEnoughItems-9.1.580
travelers-backpack-1.19.2-8.2.10
trinkets-3.4.1
youre-in-grave-danger-1.4.7

I enabled the trinket integration in Travelersbackpack config and disabled the keep Trinkets on death in yigd config.

Result:
Game acts like it doesn't even exist in the trinket slot. So it doesn't generate the grave. I tried to only place the backpack in the trinket, then just in the inventory, and then both. It generated the grave only in cases when I had SOMETHING in the inventory, but it always deleted the backpack from the trinket slot. 

NOTE: I tried enabling "generateEmtpyGraves" but even though it generated the grave, it didn't retrieve the backpack.

Configs (sorry, it doesn't let me post it as file):
<details>
<summary>Travelers Backpack</summary>

{
	"backpackSettings": {
		"disableCrafting": false,
		// Enables wearing backpack directly from ground
		"enableBackpackBlockQuickEquip": true,
		// Backpack immune to any damage source (lava, fire), can't be destroyed, never disappears as floating item
		"invulnerableBackpack": true,
		"tanksCapacity": 324000,
		// Places backpack at place where player died
		"backpackDeathPlace": true,
		// Places backpack at place where player died, replacing all blocks that are breakable and do not have inventory (backpackDeathPlace must be true in order to work)
		"backpackForceDeathPlace": false,
		"enableSleepingBagSpawnPoint": false,
		/* If true, backpack can only be worn by placing it in curios 'Back' slot
		   WARNING - Remember to TAKE OFF BACKPACK BEFORE enabling or disabling this integration!! - if not you'll lose your backpack
		*/
		"trinketsIntegration": true
	},
	"world": {
		// Enables backpacks spawning in loot chests
		"enableLoot": true
	},
	"abilities": {
		"enableBackpackAbilities": true,
		"forceAbilityEnabled": false
	},
	"slownessDebuff": {
		// Player gets slowness effect, if carries too many backpacks in inventory
		"tooManyBackpacksSlowness": true,
		// Maximum number of backpacks, which can be carried in inventory, without slowness effect
		"maxNumberOfBackpacks": 3,
		"slownessPerExcessedBackpack": 1
	},
	"client": {
		// Enables tool cycling via keybind (Default Z) + scroll combination, while backpack is worn
		"enableToolCycling": true,
		// Enables tip, how to obtain a backpack, if there's no crafting recipe for it
		"obtainTips": true,
		// Render tools in tool slots on the backpack, while worn
		"renderTools": true,
		// Render backpack if elytra is present
		"renderBackpackWithElytra": true,
		// Disable backpack rendering
		"disableBackpackRender": false,
		"overlay": {
			// Enables tanks and tool slots overlay, while backpack is worn
			"enableOverlay": true,
			// Offsets to left side
			"offsetX": 20,
			// Offsets to up
			"offsetY": 30
		}
	}
}
</details>

<details>
<summary>YIGD</summary>

[graveSettings]
generateGraves = true
retrievalType = "ON_USE"
compactRetrieveItems = true
retrievalRequireShovel = false
dropType = "IN_INVENTORY"
dropPlayerHead = true
dropGraveBlock = false
requireGraveItem = false
generateEmptyGraves = false
ignoreSpawnProtection = true
deathInSpawnProtection = "CREATE_GRAVE"
priority = "GRAVE"
unlockableGraves = true
deleteEnchantments = ["minecraft:vanishing_curse", "vanishing_curse"]
soulboundEnchantments = ["yigd:soulbound"]
loseSoulboundLevelOnDeath = false
voidSlots = []
soulboundSlots = []
applyBindingCurse = true
useLastGroundPos = false
trySoft = false
trySoftApproach = "RADIUS"
tryStrict = true
replaceWhenClaimed = true
persistGraves = false
blacklistDimensions = []
graveInVoid = true
ignoreDeathTypes = []
graveSpawnHeight = 2
lastResort = "DROP_ITEMS"
putXpInGrave = true
defaultXpDrop = false
xpDropPercent = 50
tellDeathPos = true
maxGraveBackups = 50

[graveSettings.graveRobbing]
enableRobbing = true
onlyMurderer = false
afterTime = 1
timeType = "HOURS"
robPriority = "INVENTORY"
tellRobber = true
notifyWhenRobbed = true

[graveSettings.graveDeletion]
canDelete = false
afterTime = 1
timeType = "HOURS"
dropInventory = false

[graveSettings.itemLoss]
enableLoss = false
ignoreSoulboundItems = true
affectStacks = false
usePercentRange = false
lossRangeFrom = 0
lossRangeTo = 5
percentChanceOfLoss = 100

[graveSettings.randomSpawnSettings]
percentSpawnChance = 0
spawnEntity = "minecraft:zombie"
spawnNbt = "{ArmorItems:[{},{},{},{id:\"minecraft:player_head\",tag:{SkullOwner:{Name:\"${name}\",Id:\"${uuid}\"}},Count:1b}]}"

[graveSettings.blockUnderGrave]
generateBlockUnder = true
inOverWorld = "minecraft:cobblestone"
inNether = "minecraft:soul_soil"
inTheEnd = "minecraft:end_stone"
inCustom = "minecraft:dirt"
generateWhenProtected = false

[graveSettings.graveCompatConfig]
levelzXpInGraves = true
levelzXpDropPercent = 100
claimRuleOverride = "CREATE_GRAVE"
prioritiseTheGraveyardGraves = false
graveyardSearchRadius = 10
keepAllTrinkets = false

[graveSettings.graveRenderSettings]
useRenderFeatures = true
useSpecialBlockRenderer = true
renderGraveSkull = true
renderGraveOwner = true
textShadow = true
adaptRenderer = false
glowingGrave = true
glowMinDistance = 0
glowMaxDistance = 10

[graveSettings.graveRenderSettings.guiTextColors]
graveSelectGraveLocation = 13369548
graveSelectItemSize = 204
graveSelectLevelSize = 2725384
graveSelectTitle = 5592405
graveSelectPageView = 30464
graveSelectShowAvailableCheckbox = 7829367
graveSelectShowClaimedCheckbox = 7829367
graveSelectShowDestroyedCheckbox = 7829367
graveSelectShowStatusCheckbox = 7829367
graveViewLockGrave = 0
graveViewUnlockGrave = 0
graveViewRestoreGrave = 0
graveViewDeleteGrave = 0
graveViewRobGrave = 0
graveViewGiveKey = 0
graveViewDeathMessage = 11141120
graveViewDeathDimension = 21956
graveViewCoordinates = 12255419
graveViewModItemSize = 5592405
graveViewLevelSize = 2725384
claimedBy = 0
playerSelectPlayerName = 16384
playerSelectGraveCount = 5592405
playerSelectTitle = 5592405
playerSelectPageView = 30464
playerSelectIncludeAvailableCheckbox = 7829367
playerSelectIncludeClaimedCheckbox = 7829367
playerSelectIncludeDestroyedCheckbox = 7829367
playerSelectShowZeroCheckbox = 7829367
playerSelectSearchBar = 16777215
[utilitySettings.soulboundEnchant]
enabled = true
isTreasure = false
villagerTrade = true
tableAndLoot = false
[utilitySettings.deathSightEnchant]
enabled = false
isTreasure = true
villagerTrade = true
tableAndLoot = true
[utilitySettings.scrollItem]
scrollType = "DISABLED"
retrieveOnRespawn = true
[utilitySettings.graveKeySettings]
enableKeys = false
alwaysRequire = false
retrieveOnRespawn = true
getFromGui = false
rebindable = false
graveKeySpecification = "PLAYER"
[utilitySettings.graveCompassSettings]
receiveOnDeath = false
tryDeleteOnClaim = true
[commandToggles]
coreCommandName = "yigd"
retrieveGrave = true
robGrave = true
selfView = true
adminView = true
moderateGraves = true
clearGraveBackups = true
whitelist = true
whitelistAdd = true
whitelistRemove = true
whitelistToggle = true
coordinateToggle = true
[debugConfig]
clearInventoryOnRespawn = false
createGraveBeforeDeathMessage = false

</details>

I will now try to add another mod that add trinkets to test if they get deleted and post in a separate comment.

**Comment by neonishere** [comment]
I tested with previous mods plus new ones, which are:

things-0.2.20+1.19
shulkerboxslot-fabric-5.0.0-beta.1+1.19.2
extraalchemy-fabric-1.19.2-1.9.0
Patchouli-1.19.2-77-FABRIC
owo-lib-0.9.3+1.19

I took items from every mods and put them in each trinket slot (shulker in backpack slot too)
Configs are unchanged.

Result:
Every item was returned into their place.

So, I'm guessing it's a problem on their end?

I'll retry with older version and try to install new versions again of yigd and TB, there might have been a mix up on my end

**Comment by neonishere** [comment]
Fixed when I tried with 1.4.8 YIGD. Not sure what happened here.

I think you can close the issue at this point

EDIT: I tried again with 1.4.7 and it doesn't work for some reason. But since 1.4.8 works, it doesn't matter i guess

**Comment by B1n-ry** [comment]
What? I changed like 3 lines of code in that update. Nothing changed had anything to do with travelers backpack at least. I tested myself with the mods mentioned in your first message since, well, I wanted to confirm the issue and make sure it was actually patched. Couldn't reproduce it still
Anyhow, apparently it fixed itself. I don't think this was because of any other mod than YIGD. The trinket compat works differently from the travelers backpack compat, and what's happened in the past is that since the backpack tries to place itself, it tells trinkets to delete the item on death (and YIGD checks the same thing when saving trinkets). In 1.4.7 I made YIGD intervene and tell travelers backpack it should only place the backpack on the ground if the mod is configured that way **and** graves are disabled.
I don't really know what has happened. Perhaps it could be a hardware or java version issue.
I want to thank you for taking time to research the different reproducable steps, and letting me know it got fixed somehow in 1.4.8.
I'll close the issue, but let me know if the issue would resurface, or if there'd be anything else that would be acting up.
